examples/text-classification/run_pl_glue.py

```diff
@@ -118,78 +208,129 @@ def _eval_end(self, outputs):
             "than this will be truncated, sequences shorter will be padded.",
         )
-
         parser.add_argument(
-            "--task", default="", type=str, required=True, help="The GLUE task to run",
+            "--task", type=str, required=True, help="The GLUE task to run",
         )
-
         parser.add_argument(
-            "--data_dir",
-            default=None,
-            type=str,
-            required=True,
-            help="The input data dir. Should contain the training files for the CoNLL-2003 NER task.",
+            "--data_dir", default="./glue_dir", type=str, help="Directory to save/load processed cache data"
         )
-
         parser.add_argument(
             "--overwrite_cache", action="store_true", help="Overwrite the cached training and evaluation sets"
         )
-
         return parser
 
 
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-    add_generic_args(parser, os.getcwd())
-    parser = GLUETransformer.add_model_specific_args(parser, os.getcwd())
-    args = parser.parse_args()
+def parse_args(args=None):
+    parser = ArgumentParser()
+
+    # add some script specific args
+    parser.add_argument("--seed", type=int, default=42)
+    parser.add_argument(
+        "--output_dir", type=str, default="", help="Directory to write outputs to or load checkpoint from"
+    )
+    parser.add_argument("--do_train", action="store_true", help="Run training loop")
+    parser.add_argument("--do_predict", action="store_true", help="Run test loop")
+
+    # enable all trainer args
+    parser = Trainer.add_argparse_args(parser)
+
+    # add the base module args
+    parser = BaseTransformer.add_model_specific_args(parser)
 
-    # If output_dir not provided, a folder will be generated in pwd
-    if args.output_dir is None:
-        args.output_dir = os.path.join("./results", f"{args.task}_{time.strftime('%Y%m%d_%H%M%S')}",)
-        os.makedirs(args.output_dir)
+    # add the glue module args
+    parser = GLUETransformer.add_model_specific_args(parser)
 
+    # cook them all up :)
+    args = parser.parse_args(args)
+
+    return args
+
+
+def main(args):
+
+    # Init Model
     model = GLUETransformer(args)
-    trainer = generic_train(model, args)
 
-    # Optionally, predict on dev set and write to output_dir
+    # Init Trainer
+    trainer = Trainer.from_argparse_args(args)
+
+    # Include custom callback for logging results
+    trainer.callbacks.append(LoggingCallback())
+
+    # Run training and reload best model from checkpoint
+    if args.do_train:
+
+        # Fit the model
+        trainer.fit(model)
+
+        # Reload best model from current experiment run's checkpoint directory
+        experiment_ckpt_dir = model.trainer.weights_save_path
+        checkpoints = list(sorted(glob.glob(os.path.join(experiment_ckpt_dir, "epoch=*.ckpt"), recursive=True)))
+        trainer.resume_from_checkpoint = checkpoints[-1]
```

Comment:
can we test/assert that this change actually causes a new state dict to be loaded? or get some input from @williamFalcon ?

Reply:
Very good point. I've never seen anybody do what I did here and I didn't test it myself...just kinda assumed when `trainer.test` or `trainer.fit` are called, it would look to see what is set for `trainer.resume_from_checkpoint` and load if need be.

Reply:
confirmed it works